modules/integration/tests-integration/tests-amqp/src/test/java/org/wso2/mb/integration/tests/amqp/functional/MultiTenantTopicTestCase.java

```diff
@@ -0,0 +1,161 @@
+
+        AndesClient tenant1ReceivingClient1 = new AndesClient("receive", "127.0.0.1:5672", "topic:topictenant1.com/tenantTopic",
+                "100", "false", Integer.toString(runTime), Integer.toString(expectedMessageCount),
+                "1", "listener=true,ackMode=1,delayBetweenMsg=0,stopAfter=" + expectedMessageCount, "",
+                "topictenantuser1!topictenant1.com", "topictenantuser1");
+        tenant1ReceivingClient1.startWorking();
+
+        AndesClient tenant1ReceivingClient2 = new AndesClient("receive", "127.0.0.1:5672", "topic:topictenant1.com/tenantTopic",
+                "100", "false", Integer.toString(runTime), Integer.toString(expectedMessageCount),
+                "1", "listener=true,ackMode=1,delayBetweenMsg=0,stopAfter=" + expectedMessageCount, "",
+                "topictenantuser2!topictenant1.com", "topictenantuser2");
+        tenant1ReceivingClient2.startWorking();
+
+        // Start sending client (tenant1)
+        AndesClient tenant1SendingClient = new AndesClient("send", "127.0.0.1:5672", "topic:topictenant1.com/tenantTopic",
+                "100", "false", Integer.toString(runTime), Integer.toString(sendMessageCount), "1",
+                "ackMode=1,delayBetweenMsg=0,stopAfter=" + sendMessageCount, "", "topictenantuser1!topictenant1.com", "topictenantuser1");
+
+        tenant1SendingClient.startWorking();
+
+
+        boolean tenant1ReceiveSuccess1 = AndesClientUtils.waitUntilMessagesAreReceived(tenant1ReceivingClient1,
+                expectedMessageCount, runTime);
+        boolean tenant1ReceiveSuccess2 = AndesClientUtils.waitUntilMessagesAreReceived(tenant1ReceivingClient2,
+                expectedMessageCount, runTime);
+        boolean adminReceiveSuccess = AndesClientUtils.waitUntilMessagesAreReceived(adminReceivingClient,
+                expectedMessageCount, runTime);
+
+        boolean tenant1SendSuccess = AndesClientUtils.getIfSenderIsSuccess(tenant1SendingClient, sendMessageCount);
+
+        Assert.assertTrue(tenant1SendSuccess, "Sending failed for tenant 1 user 1.");
+        Assert.assertTrue(tenant1ReceiveSuccess1, "Message receiving failed for tenant 1 user 1.");
+        Assert.assertTrue(tenant1ReceiveSuccess2, "Message receiving failed for tenant 1 user 2.");
+        Assert.assertEquals(adminReceivingClient.getReceivedTopicMessagecount(), expectedMessageCount);
+        Assert.assertTrue(adminReceiveSuccess, "Message receiving failed for admin of tenant 1.");
+
+    }
+
+    /**
+     *
+     * Test case 2
+     * 1. Start 2 subscribers from different tenant for the same topic
+     * 2. Start 2 publishers from different tenant for the same topic
+     * 3. Subscribers should receive the message from their tenant only
+     */
+    @Test(groups = "wso2.mb", description = "Multiple Tenant Single Users Test")
+    public void performMultipleTenantTopicTestCase() {
+        int sendMessageCount = 100;
+        int runTime = 20;
+        int expectedMessageCount = 200;
```

Comment:
Shall we make two different expected message counts and publish different number of messages just in case messages are mixed up.